hw3/02-charts.js

```diff
@@ -29,24 +31,123 @@ const borderColors = [
 // url for the Thrones API
 const url = 'https://thronesapi.com/api/v2/Characters';
 
-const renderChart = () => {
+
+// async function to fetch the data from a given url
+const fetchData = async function fetchDataFromApi(url) {
+  try {
+      const res = await fetch(url);
+      const data = await res.json();
+      return data;
+  }
+  catch {
```

Comment:
looks like catch doesn't catch the error.

Reply:
I did this often through the code, but i have it resolved or most of them in my next. commit